### ISSUE TITLE
Upgrade from C121 -> C122 on macOS x64 via 1.64.x - Nightly (Checking C122)

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#35908]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [ ] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.64.54 Chromium: 122.0.6261.29 (Official Build) nightly (x86_64)
-- | --
Revision | cf48cb2a6c35ed530efd5b15349ef4ee81301594
OS | macOS Version 13.6.4 (Build 22G513)
```

**Comment by LaurenWags** [comment]
Upgrade checks in progress:

```

```

<details>
<summary>C122 --> C122 on-demand upgrade check - InProgress</summary>

1. Install `C122` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1707427634501499?thread_ts=1707427626.588229&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

7. Confirm you are prompted to "Relaunch" to get update (note, if you click "Relaunch" button you will relaunch and could potentially pick up a newer Nightly that is on the previous Chromium version. Closing and then relaunching via the terminal will not update you again)

SS

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------


</details>


<details>
<summary>C121 --> C122 on-demand upgrade check - TBD</summary>

1. Install `C121` build via TBD
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/nightly/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

7. Confirm you are prompted to "Relaunch" to get update (see note in first case about this button)

SS

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------


</details>

<details>
<summary>C122 --> C122 background upgrade check - TBD</summary>

1. Install `C122` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1707427634501499?thread_ts=1707427626.588229&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`

SS

</details>

<details>
<summary>C121 --> C122 background upgrade check - TBD</summary>

1. Install `C121` build via TBD
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`

SS

</details>

**Comment by LaurenWags** [comment]
Upgrade checks (using .pkg) in progress:

```
Brave | 1.64.54 Chromium: 122.0.6261.29 (Official Build) nightly (x86_64)
-- | --
Revision | cf48cb2a6c35ed530efd5b15349ef4ee81301594
OS | macOS Version 13.6.4 (Build 22G513)
```

and

```
Brave	1.64.57 Chromium: 122.0.6261.29 (Official Build) nightly (x86_64) 
Revision	7d57398776cbf95dafe0c6544e0a64eb9f278379
OS	macOS Version 13.6.4 (Build 22G513)
```

<details>
<summary>C122 --> C122 background upgrade check - InProgress</summary>

1. Install `C122` PR .pkg build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1707427634501499?thread_ts=1707427626.588229&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
8. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
9. DO NOT navigate to `brave://settings/help`
10. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

11. Close and restart the browser via terminal with same command as step 7
12. Check version is updated under `brave://version`

SS

Confirmed delta for this upgrade:

**`Example`** | **`Example`**
----------------|----------------

</details>

<details>
<summary>C121 --> C122 background upgrade check - TBD</summary>

1. Install `C121` build via https://github.com/brave/brave-browser/releases/tag/v1.64.54
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
9. DO NOT navigate to `brave://settings/help`
10. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

11. Close and restart the browser via terminal with same command as step 7
12. Check version is updated under `brave://version`

SS

</details>

**Comment by LaurenWags** [comment]
Upgrade checks (using arm64 .dmg):

```

```

and

```

```

<details>
<summary>C122 --> C122 on-demand upgrade check - InProgress</summary>

1. Install `C122` PR build (arm64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1707427634501499?thread_ts=1707427626.588229&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
6. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
7. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

8. Confirm you are prompted to "Relaunch" to get update (note, if you click "Relaunch" button you will relaunch and could potentially pick up a newer Nightly that is on the previous Chromium version. Closing and then relaunching via the terminal will not update you again)

SS

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------


Confirmed delta for this upgrade:

**`Example`** | **`Example`**
----------------|----------------


</details>


<details>
<summary>C121 --> C122 on-demand upgrade check - TBD</summary>

1. Install `C121` build via https://github.com/brave/brave-browser/releases/tag/v1.64.54
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/nightly/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

7. Confirm you are prompted to "Relaunch" to get update (see note in first case about this button)

SS

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------


</details>

<details>
<summary>C122 --> C122 background upgrade check - TBD</summary>

1. Install `C122` PR build (arm64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1707427634501499?thread_ts=1707427626.588229&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
8. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
9. DO NOT navigate to `brave://settings/help`
10. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

11. Close and restart the browser via terminal with same command as step 7
12. Check version is updated under `brave://version`

SS

Confirmed delta for this upgrade:

**`Example`** | **`Example`**
----------------|----------------


</details>

<details>
<summary>C121 --> C122 background upgrade check - TBD</summary>

1. Install `C121` build via https://github.com/brave/brave-browser/releases/tag/v1.64.54
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
9. DO NOT navigate to `brave://settings/help`
10. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

11. Close and restart the browser via terminal with same command as step 7
12. Check version is updated under `brave://version`

SS

</details>
